packages/explorer/src/queries/dozer/useTableDataQuery.ts

```diff
@@ -0,0 +1,53 @@
+import { useParams } from "next/navigation";
+import { Schema } from "@latticexyz/config";
+import { useQuery } from "@tanstack/react-query";
+import { useDozerUrl } from "../../hooks/useDozerUrl";
+
+type DozerResponse = {
+  block_height: number;
+  result: [string[][]];
+};
+
+type Props = {
+  schema?: Schema;
+  query?: string;
+};
+
+export function useTableDataQuery({ schema, query }: Props) {
+  const { worldAddress, chainName } = useParams();
+  const dozerUrl = useDozerUrl();
+
+  return useQuery({
+    queryKey: ["table", worldAddress, chainName, query],
+    queryFn: async () => {
+      const response = await fetch(dozerUrl, {
```

Comment:
broad question, not strictly tied to this PR: have we fiddled with server actions instead of this react-query+fetch pattern?

Reply:
interesting question I don't have a definite answer to yet. Even if we were to use server actions, react-query is helpful for polling. Another point to consider is that server-actions are blocking meaning they can't run in parallel. For the current setup I don't see that as a problem but could be down the line.

I was meaning to dive deeper into server actions, specifically the way they behave and impact behavior in next.js, to understand exactly where it could be useful. Just for the sake of type-safety I'd love to incorporate it if it actually makes sense but still need to do more research. For now the good old API is a battle-tested solution in my eyes.